app/src/main/java/cloudcity/util/CellUtil.java

```diff
@@ -65,18 +67,60 @@ public static int remapCellClassTypeIntoInteger(CellInformation cellToRemap) {
         return retVal;
     }
 
+    public static @NonNull CellInfoModel getCellInfoModel(CellInformation currentCell) {
+        CellInfoModel cellInfoModel = new CellInfoModel();
+
+        // Looking at the OMNT code, only CDMA doesn't have 'bands', GSM,LTE and NR all have it
+        if (currentCell instanceof CDMAInformation) {
+            /* No information in 2G and 3G set dummy data. */
+            cellInfoModel.setDummy(1);
+        } else {
+            /* Real data available for all other network types. */
+            // Ok so bands is technically the ARFCN
+            String bandsString = currentCell.getBands();
+            // Clean up the bandsString, which will come as [x], so we need to get rid of the brackets
+            // and then parse the number inside of it.
+            String sanitizedBandsString = bandsString.replaceAll("\\[","").replaceAll("]","");
+            if (!sanitizedBandsString.isBlank()) {
+                // Seems like this "[]" is what's being returned on SIM-less phones (or on WiFi)
+                // So try parsing only if the sanitized string is non-blank
+                int arfcn = Integer.parseInt(sanitizedBandsString);
+                cellInfoModel.setEarfcn(arfcn);
+            } else {
+                //TODO replace with CloudCityLogger when that's merged in (or on that other branch)
+                Log.e(TAG, "Error parsing bands (EARFCN) since it wasn't present!");
+            }
+            cellInfoModel.setPci(currentCell.getPci());
+        }
+
+        long id = currentCell.getCi();
+
+        if (currentCell instanceof NRInformation) {
+            if (id != CellInfo.UNAVAILABLE_LONG) {
+                cellInfoModel.setCellId(currentCell.getCi());
+            }
+        } else if (currentCell instanceof LTEInformation) {
+            if (id != CellInfo.UNAVAILABLE) {
+                cellInfoModel.setCellId((int)(currentCell.getCi() >> 8));
+                cellInfoModel.seteNodeBId((int) currentCell.getCi() & 0x000000FF);
+            }
+        }
+
+        return cellInfoModel;
+    }
+
     /**
      * Handy method for doint everything we normally do manually, find the registered cell, grab the signal strength
      * and then init the {@link MeasurementsModel} from the registered cell and update with the signal strength
      *
      * @param dp the {@link DataProvider} to grab all of these things from
      * @return the MeasurementsModel to be uploaded
      */
-    public static MeasurementsModel getRegisteredCellInformationUpdatedBySignalStrengthInformation(@NonNull DataProvider dp) {
+    public static Pair<MeasurementsModel, CellInfoModel> getRegisteredCellInformationUpdatedBySignalStrengthInformation(@NonNull DataProvider dp) {
```

Comment:
this is changed to `Pair` since both of these are obtained from the currently-registered cell, but both of them are members of the *concrete* NetworkDataModel, be it MobileSignalNetworkDataModel or the Iperf3NetworkDataModel.

since they're both members, and this is a convenient place to calculate and capture both of them - I figured it'd be best to return both from *this* method rather than making this same exact code (ok, similar) to return just the other thing.

for now this is just used for sending the iperf3 data but maybe we decide to use it differently in the future. (also, another clarifying comment, nothing actionable here)